repositories/companyRepository.go

```diff
@@ -20,58 +20,152 @@ func (repository *CompanyRepository) GetAllCompanies() ([]models.Company, error)
+		})
 	}
 	rows.Close()
 	return companies, nil
 }
 
 func (repository *CompanyRepository) GetCompany(id string) (models.Company, error) {
 	var company models.Company
-	err := repository.DB.QueryRow(context.Background(), "select * from companies where id=$1", id).Scan(&company.Id, &company.Name, &company.IsMain)
+
+	Uuid, err := uuid.FromString(id)
+	if err != nil {
+		return company, err
+	}
+
+	rows, err := repository.DB.Query(context.Background(), `select * from companies where id = $1`, Uuid)
+	defer rows.Close()
+
+	if err != nil {
+		return company, err
+	}
+
+	if !rows.Next() {
+		return company, errors.New("There is no company with this id")
+	}
+
+	var companyPers persistence.Companies
+	companyPers.Scan(&rows)
+
+	var stringUUID string
+	err = companyPers.Id.AssignTo(&stringUUID)
 	if err != nil {
 		return company, err
 	}
+
+	company = models.Company{
+		Id:     stringUUID,
+		Name:   companyPers.Name,
+		IsMain: companyPers.Ismain,
+	}
+
 	return company, nil
 }
 
 func (repository *CompanyRepository) AddCompany(company *models.Company) error {
-	u := uuid.NewV4()
-	company.Id = u.String()
-	_, err := repository.DB.Exec(context.Background(), "insert into public.companies (id, name, ismain) values ($1, $2, $3)", u.Bytes(), company.Name, company.IsMain)
+	tx, err := repository.DB.Begin(context.Background())
 	if err != nil {
 		return err
 	}
+	defer tx.Rollback(context.Background())
+
+	company.Id = uuid.NewV4().String()
+	companyPers := persistence.Companies{
+		Name:   company.Name,
+		Ismain: company.IsMain,
+	}
+	companyPers.Id.Set(company.Id)
+
+	_, err = companyPers.InsertTx(&tx)
+	if err != nil {
+		return err
+	}
+
+	tx.Commit(context.Background())
+
 	return nil
 }
 
 func (repository *CompanyRepository) UpdateCompany(company models.Company) error {
-	commandTag, err := repository.DB.Exec(context.Background(),
-		"UPDATE public.companies SET name=$1, ismain=$2 WHERE id=$3",
-		company.Name, company.IsMain, company.Id)
+	tx, err := repository.DB.Begin(context.Background())
 	if err != nil {
 		return err
 	}
-	if commandTag.RowsAffected() != 1 {
+	defer tx.Rollback(context.Background())
+
+	companyPers := persistence.Companies{
+		Name:   company.Name,
+		Ismain: company.IsMain,
+	}
+	companyPers.Id.Set(company.Id)
+
+	commandTag, err := companyPers.UpdateTx(&tx)
+	if err != nil {
+		return err
+	}
+	if commandTag != 1 {
 		return errors.New("No row found to update")
 	}
+
+	tx.Commit(context.Background())
 	return nil
 }
 
 func (repository *CompanyRepository) DeleteCompany(id string) error {
-	commandTag, err := repository.DB.Exec(context.Background(), "DELETE FROM public.companies WHERE id=$1;", id)
+	tx, err := repository.DB.Begin(context.Background())
 	if err != nil {
-		fmt.Println(err)
 		return err
 	}
-	if commandTag.RowsAffected() != 1 {
+	defer tx.Rollback(context.Background())
+
+	companyPers := persistence.Companies{}
+	companyPers.Id.Set(id)
+
+	commandTag, err := companyPers.DeleteTx(&tx)
```

Comment:
To avoid some code repetition, you could make a method which handles this case:
if err != nil {
		return err
	}
	if commandTag != 1 {
		return errors.New("No row found to delete")
	}